packages/aws-cdk/test/integ/cli/cdk-helpers.ts

```diff
@@ -2,8 +2,99 @@ import * as child_process from 'child_process';
 import * as fs from 'fs';
 import * as os from 'os';
 import * as path from 'path';
-import { outputFromStack, testEnv, AwsClients } from './aws-helpers';
-import { TestEnvironment } from './test-helpers';
+import { outputFromStack, AwsClients } from './aws-helpers';
+import { ResourcePool, ILease } from './resource-pool';
+import { TestContext } from './test-helpers';
+
+const REGIONS = process.env.AWS_REGIONS
+  ? process.env.AWS_REGIONS.split(',')
+  : [process.env.AWS_REGION ?? process.env.AWS_DEFAULT_REGION ?? 'us-east-1'];
+
+process.stdout.write(`Using regions: ${REGIONS}\n`);
+
+const REGION_POOL = new ResourcePool(REGIONS);
+
+
+export type AwsContext = { readonly aws: AwsClients };
+
+/**
+ * Higher order function to execute a block with an AWS client setup
+ *
+ * Allocate the next region from the REGION pool and dispose it afterwards.
+ */
+export function withAws<A extends TestContext>(block: (context: A & AwsContext) => Promise<void>) {
+  return (context: A) => REGION_POOL.using(async (region) => {
+    const aws = await AwsClients.forRegion(region, context.output);
+    await sanityCheck(aws);
+
+    return block({ ...context, aws });
+  });
+}
+
+/**
+ * Higher order function to execute a block with a CDK app fixture
+ *
+ * Requires an AWS client to be passed in.
+ *
+ * For backwards compatibility with existing tests (so we don't have to change
+ * too much) the inner block is expecte to take a `TestFixture` object.
+ */
+export function withCdkApp<A extends TestContext & AwsContext>(block: (context: TestFixture) => Promise<void>) {
```

Comment:
Very minor: Not just "with cdk app" but also "with region-based leases". 
Could maybe split that into its own higher order function chain-link, or move that within `withDefaultFixture()`.

Reply:
Well this wrapper doesn't ADD a region based lease, it adds a cdk fixture.

It does REQUIRE a region lease to do that, but that's not what the name means.

Reply:
The idea being that these wrappers should be freely composeable to add more setup and teardown.